apps/judicial-system/backend/src/app/modules/notification/subpoenaNotification.service.ts

```diff
@@ -0,0 +1,222 @@
+import {
+  Inject,
+  Injectable,
+  InternalServerErrorException,
+} from '@nestjs/common'
+import { InjectModel } from '@nestjs/sequelize'
+
+import { IntlService } from '@island.is/cms-translations'
+import { EmailService } from '@island.is/email-service'
+import { type Logger, LOGGER_PROVIDER } from '@island.is/logging'
+import { type ConfigType } from '@island.is/nest/config'
+
+import { INDICTMENTS_COURT_OVERVIEW_ROUTE } from '@island.is/judicial-system/consts'
+import { NotificationType } from '@island.is/judicial-system/types'
+
+import { CaseService } from '../case'
+import { EventService } from '../event'
+import { SubpoenaService } from '../subpoena'
+import { DeliverResponse } from './models/deliver.response'
+import { Notification, Recipient } from './models/notification.model'
+import { BaseNotificationService } from './baseNotification.service'
+import { notificationModuleConfig } from './notification.config'
+import { strings } from './subpoenaNotification.strings'
+
+@Injectable()
+export class SubpoenaNotificationService extends BaseNotificationService {
+  constructor(
+    @InjectModel(Notification)
+    notificationModel: typeof Notification,
+    @Inject(notificationModuleConfig.KEY)
+    config: ConfigType<typeof notificationModuleConfig>,
+    @Inject(LOGGER_PROVIDER) logger: Logger,
+    intlService: IntlService,
+    emailService: EmailService,
+    eventService: EventService,
+    private readonly caseService: CaseService,
+    private readonly subpoenaService: SubpoenaService,
+  ) {
+    super(
+      notificationModel,
+      emailService,
+      intlService,
+      config,
+      eventService,
+      logger,
+    )
+  }
+
+  private async sendServiceSuccessfulNotification(
```

Comment:
We might need to keep in mind that we should only send these notifications once, are we handling that anywhere? We should only have to send these the first time the status is updated. 
While it's true that technically we shouldn't ever get here twice, we should keep in mind that since the initial change is coming from an external system it may behave weirdly and send us info we've already received.

Reply:
There are helper functions that can help with this. In `InternalNotificationService` you will find `hasSentNotification` and `hasReceivedNotification`. These could be moved to the `BaseNotificationService` and used here.